static/js/publisher/form/banner.js

```diff
@@ -0,0 +1,260 @@
+import React, { Fragment } from "react";
+import PropTypes from "prop-types";
+
+class Banner extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.bannerImageChangeHandler = this.bannerImageChangeHandler.bind(this);
+    this.bannerIconChangeHandler = this.bannerIconChangeHandler.bind(this);
+    this.handleChangeImageClick = this.handleChangeImageClick.bind(this);
+    this.handleChangeIconClick = this.handleChangeIconClick.bind(this);
+    this.handleRemoveImageClick = this.handleRemoveImageClick.bind(this);
+    this.handleRemoveIconClick = this.handleRemoveIconClick.bind(this);
+
+    this.state = {
+      bannerImage: this.props.bannerImage,
+      bannerIcon: this.props.bannerIcon
+    };
+  }
+
+  componentDidUpdate() {
+    const images = [];
+    if (this.state.bannerImage.url) {
+      images.push(this.state.bannerImage);
+    }
+    if (this.state.bannerIcon.url) {
+      images.push(this.state.bannerIcon);
+    }
+
+    this.props.updateImageState(images);
+  }
+
+  bannerImageChangeHandler() {
+    const file = this.bannerImageInput.files[0];
+    if (file.name !== "banner.jpg" && file.name !== "banner.png") {
```

Comment:
The file name restriction is purely internal, because screenshots are used to store them.
It's a pity that having a dedicated UI we still expose this internal limitation to the user by forcing them to name their images properly.

Could we rename the file on our front- or backend to the names required by the store?

Reply:
We're in discussions on how we're going to improve banners (to be used across both the website and desktop store) this is an interim implementation.

I've added [this issue](https://github.com/canonical-websites/snapcraft.io/issues/1698) so we can discuss it further outside this PR :+1: